DATAMONGO-2016 fixing issue where question mark in password broke option extraction

steve-thousand opened the pull request:
I tried to follow the guidelines, but I can't seem to figure out how to create a bug for this issue in the JIRA. This is actually my first attempt at a contribution so I'm sure I'm missing something. I did search the JIRA for an issue with credentials and "?" characters but couldn't find one.

The issue I've run into is that if a mongo password contains a "?" character, but no auth-mechanism option, then the MongoCredentialPropertyEditor will throw an ArrayIndexOutOfBoundsException. I've done some research on mongo password rules and there doesn't seem to be any limit on using a "?" in the password.

- [X] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAMONGO). **was not sure how to actually create an issue**
- [X] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [X] You submit test cases (unit or integration tests) that back your changes.
- [X] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

pivotal-issuemaster commented:
@steve-thousand Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-mongodb&pullRequestId=578)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-mongodb&pullRequestId=578) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

steve-thousand commented on spring-data-mongodb/src/main/java/org/springframework/data/mongodb/config/MongoCredentialPropertyEditor.java:
```diff
@@ -164,7 +165,7 @@ private static String extractDB(String text) {
 
 		int optionsSeparationIndex = text.lastIndexOf(OPTIONS_DELIMITER);
-		int dbSeparationIndex = text.lastIndexOf(OPTIONS_DELIMITER);
+		int dbSeparationIndex = text.lastIndexOf(DATABASE_DELIMITER);
```
this line in the source jumped out at me as a possible copy/paste error. changing the delimiter we check for fixed the issue

pivotal-issuemaster commented:
@steve-thousand Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-mongodb&pullRequestId=578)!

mp911de commented:
I filed https://jira.spring.io/browse/DATAMONGO-2016 to track this issue. Please go to https://jira.spring.io/secure/Signup!default.jspa and create a user account so we can collaborate on the ticket. Please attach a stack trace to the mentioned ticket as comment.

steve-thousand commented:
Thanks @mp911de I've attached the stacktrace to the ticket, added the ticket number to the test I added, and updated the pull request title to include the ticket number.

mp911de commented:
Thanks a lot for your contribution. That's merged, polished, and backported now.
